test/excludes/SchemaDumperTest.rb

```diff
@@ -1,2 +1,5 @@
 exclude "test_schema_dump_with_timestamptz_datetime_format", TRIAGE_MSG
 exclude "test_schema_dump_when_changing_datetime_type_for_an_existing_app", TRIAGE_MSG
+if ActiveRecord::Base.lease_connection.pool.server_version(ActiveRecord::Base.lease_connection) < 15_00_00
+  exclude "test_schema_dumps_unique_constraints", TRIAGE_MSG
+end
```

Comment:
@BuonOmo I feel like we may want to re-implement this one locally since it seems important.

@StoneGod do you know why this one was failing?

Reply:
It's failing because my local version pg is 14, but I think originally test should check version

Reply:
This test fails because `nulls_not_distinct: true` doesn't exist in the `position_4` index in the schema dump. This happens because that is only supported in PostgreSQL 15+ as noted in https://github.com/rails/rails/pull/48608 which added the capability.
I think it makes sense to skip this test if the PostgreSQL version in use happens to be lower than 15 even if we can set that version in CI, because there may be situations like @StoneGod and I experienced where we ran the tests locally in an environment that wasn't using version 15+.